src/librustdoc/html/render/cache.rs

```diff
@@ -164,14 +169,15 @@ crate fn build_index(krate: &clean::Crate, cache: &mut Cache) -> String {
     )
 }
 
-crate fn get_index_search_type(
+crate fn get_index_search_type<'tcx>(
     item: &clean::Item,
-    cache: Option<&Cache>,
+    cache: &Cache,
```

Comment:
Why not keep the optional so we don't have to keep and pass around an empty, dummy cache, since as you said `Cache` is a very large type?

Reply:
Because at some point in the tree call, it'll be required, so better instantiate it once and for all at the top.